### PR TITLE
Added Kingdom of Miscellania plugin

### PR DESCRIPTION
**Pull request by Infinitay**
This plugin adds the ability to check on your kingdom whether or not you're at your kingdom, as well as capable of alerting you when your kingdom's favor and/or coffer is low by displaying the infobox. 

[Click here to view alert demonstration](https://streamable.com/9o83m)
[Click here to view favor demonstration](https://streamable.com/ndjkd)
[Click here to view coffer demonstration](https://streamable.com/74ybb)

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomPlugin.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	public boolean isInKingdom()
+	{
+		return Ints.indexOf(client.getMapRegions(), KINGDOM_REGION) >= 0;
+	}
+
+	private void addKingdomInfobox()
+	{
+		if (counter == null)
+		{
+			counter = new KingdomCounter(itemManager.getImage(ItemID.TEAK_PRIZE_CHEST), this);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomPlugin.java]: It is not safe to getImage from the non-game thread. Maybe get the image on startup and hold it.

**Comment by Infinitay** [comment]
Everything looks good on my end. Logging in and out as well as restarting sessions, and plugin is working as intended.

Only minor _bug_ there is, is that when you relog (relog, hop, new session). Additionally, this is when your **coffer and favor is not lower** than the values you specified, **only shown in kingdom and when low** are **both enabled**. The bug adds and removes the infobox (once each) because of the **change in varbits when you logout**. Your varbits get set to 0, and depending on your settings, because they changed to 0, it could trigger your alert when low options, causing the infobox to be added. However, because you just logged in, your varbits are still changing and when the client updates the varbit to the correct value, it will trigger the removal of the infobox, once again, depending on your setting.

I say minor because I doubt and hope it does not affect performance, and only happens when you log back into the game. If someone would like to test performance that would be wonderful. 

I believe one way to mitigate this would be if VarbitChanged would support methods such as `getOldValue`, `getNewValue`, and last but not least, `getId`. With this, maybe I could add the logic to ignore the non-dependent varbits as well as not updating when logging into the game.

**Comment by Infinitay** [comment]
Currently, chat notification is non-configurable, sends message once every 5 minutes (logged in)

Chat notification preview

**Comment by Kamielvf** [comment]
Don't varbits get set to -1? And not to 0. Might be different in this case but you could have it only be valid if its `>= 0`

**Comment by Infinitay** [comment]
AFAIK, they're 0, atleast when running `Client#getSettings` within both startUp and onVarbitChanged

**Comment by Joshua-F** [comment]
When a varp is 0 then all the varbits using it are 0. Varbits are not able to be assigned a negative value, nor do they default to a negative value.

**Comment by Infinitay** [comment]
@Adam- Erm, I accidentally got rid of your commit for Notifier.java :x

I think it was this https://github.com/runelite/runelite/pull/1006/commits/20699f82cb97350a803f5223f59ea3db45050716

**Pull request merged**